.travis.yml

```diff
@@ -49,60 +19,161 @@ addons:
-  - |
-    if [[ ${TRAVIS_PHP_VERSION:0:3} == "7.0" || "${SYMFONY_VERSION}" != "" ]]; then
-      ./tests/e2e_tests bin/infection;
-    else
-      make build/bin/infection.phar;
-      ./tests/e2e_tests build/bin/infection.phar;
-    fi
-  - if [[ $NO_DEBUGGER == 1 ]]; then xdebug-disable; fi
-  - |
-    if [[ $PHPDBG != 1 ]]; then
-      bin/infection $INFECTION_FLAGS;
-    else
-      phpdbg -qrr bin/infection $INFECTION_FLAGS;
     fi
 
-after_success:
-  - bash <(curl -s https://codecov.io/bash)
-
-deploy:
-    provider: releases
-    api_key:
-        secure: JtdvJBE2ARf6VWN5KlXAburz3FZMriiXBHEdeIocHePBAzgYW7tVMksJR5N3azrE3KjhftMEukXtA1k5VfAGdLCSm6nKGsZwolkd24ryG7PYNuAdFWfbau8QQXn6aLFesF2lkTq89+PqxAdVjwXNgFA7ble3vL66Ogabff6gov6QwT9vDrTqwLmWNGwbt9QB32IE9yVPjW/3lk4MAkMHtKDr2DElyeT2CRRuPuUCA+zNwoGG95wqLAD/RwjsqR3GjotzDKrldqog9jU2OQptUc3BgdKlYR1AzdknxJYnT7a5CRVdmSOEfD9IrDqlYEzn9pHav+Sk4KD2KNKsALQGk/w16BqncN2uyOexUTzXvnDd27G4ZYy+1OsXMdYMhF5ffwj7EqNLm8XfLPtHHV0/FUJY9kkrbf9+wQVjDQmMJLLIiBSt3vuY0ya5SVW8thtjlF2j67smamNe4hDhditBipto+9Y5vI4AYhcm4CQlqP8BVbw3/LgGAl/dJR5ohM7PFZH3EUE1AojBaEXfUymlICdFVZFJbB0D/AM8jSqNmWBQu9QX3MURhT5xLSwc+kir2izQaEo/GQz79AifqkUJgSCzpRUJntCN8A7yk1uxDZDONbINF3ReuF+nqu7GMksgKZLCGbOIaBl9of24K5BeTqxgcEykiHFFY7X9QdI3Dyo=
-    file:
-        - build/bin/infection.phar
-        - build/bin/infection.phar.pubkey
-    skip_cleanup: true
-    on:
-        tags: true
-        repo: infection/infection
-        condition: "$EXECUTE_DEPLOYMENT"
+jobs:
+  include:
+    -
+      stage: Static Code Analysis
+      php: 7.2
+      install:
+        - composer install
+      script:
+        - make analyze --keep-going
+
+    - &STANDARD_TEST_JOB
+      stage: Test
+      php: 7.0
+      install:
+        - composer install
+        - |
+          if [ "${SYMFONY_VERSION}" != "" ]; then
+            composer config --unset platform
+            composer require \
+            symfony/console:${SYMFONY_VERSION} \
+            symfony/filesystem:${SYMFONY_VERSION} \
+            symfony/process:${SYMFONY_VERSION} \
+            symfony/finder:${SYMFONY_VERSION} \
+            symfony/yaml:${SYMFONY_VERSION}
+          fi
+      script:
+        - if [[ $PHPDBG != 1 ]]; then xdebug-enable; fi
+        - if [[ $PHPDBG != 1 ]]; then $PHPUNIT_BIN; else phpdbg -qrr $PHPUNIT_BIN; fi
+        - |
+          if [[ ${TRAVIS_PHP_VERSION:0:3} == "7.0" || "${SYMFONY_VERSION}" != "" ]]; then
+            ./tests/e2e_tests bin/infection;
+          else
+            make build/bin/infection.phar;
+            ./tests/e2e_tests build/bin/infection.phar;
+          fi
+        - if [[ $NO_DEBUGGER == 1 ]]; then xdebug-disable; fi
+        - |
+          if [[ $PHPDBG != 1 ]]; then
+            bin/infection $INFECTION_FLAGS;
+          else
+            phpdbg -qrr bin/infection $INFECTION_FLAGS;
+          fi
+
+      after_success:
+        - bash <(curl -s https://codecov.io/bash)
+
+    -
+      <<: *STANDARD_TEST_JOB
+      stage: Test
```

Comment:
It's not necessary to repeat this line - it is already inherited from `STANDARD_TEST_JOB`.